### ISSUE TITLE
No callback returned when Dialogflow_V2.startListening

### ISSUE THREAD

**Issue by BenjaminPcrd** [innFactory/react-native-dialogflow#63]
The callback is never returned when startListening

`
        <Button onPress={() => {
          Dialogflow_V2.onListeningStarted(()=>{
            console.log("listening started");
          });

          Dialogflow_V2.onListeningCanceled(()=>{
            console.log("listening canceled");
          });

          Dialogflow_V2.onListeningFinished(()=>{
            console.log("listening finished");
          });

          Dialogflow_V2.onAudioLevel(level=>{
            console.log(level);
          });

          Dialogflow_V2.startListening(result=>{
              console.log(result);
          }, error=>{
              console.log(error);
          });
        }} ><Text>start listening</Text></Button>

        <Button onPress={() => {
          Dialogflow_V2.finishListening()
        }}>
          <Text>Stop listening</Text>
        </Button>
`

**Comment by etcheverryRafael** [comment]
I am experiencing the same in Android. Did you find a solution? 
I have noticed the Dialogflow_V2.onListeningStarted get's called twice.
Also i think that the request is not reaching Dialogflow backend, i haved checked the console on Firebase and there is no record of any request. 
Thanks!

**Comment by BenjaminPcrd** [comment]
No, I still using the V1 for now.
A solution would be to make a speech-to-text recognition with rn-voice and next, send a text query. I'll try to make it

**Comment by etcheverryRafael** [comment]
Yes i though of that too. But it very strange that this is not working, it's the core functionality of the package.

**Comment by BenjaminPcrd** [comment]
import Voice from 'react-native-voice';
    import { Dialogflow_V2 } from 'react-native-dialogflow';
    ...

    Voice.onSpeechStart = () => console.log("Start")
    Voice.onSpeechEnd = () => console.log("End")
    Voice.onSpeechError = (err) => console.log("Error", err)

    Voice.start('en-US')

    Voice.onSpeechResults = (res) => {
      let speech = res.value[0]
      Dialogflow_V2.requestQuery(
        speech,
        result => {
          console.log(result.queryResult.fulfillmentText)
        },
        error => console.log(error)
      );
    }

That's works very well

**Comment by matheusleite** [comment]
I am experiencing the same error on Android!
Any updates?

**Comment by OscarSaro** [comment]
Same issue here in Android, the events works fine but the `startListening` callback never gets called, and the agent is not receiving anything (I have been looking though the firebase and Dialogflow logs).

The @BenjaminPcrd solution works for me, but is there any update?

**Comment by jaalbertill** [comment]
You can change de module: .../node_modules/react-native-dialogflow/index.js

AFTER:
`/**
 * DIALOGFLOW V2
 */
var dialogflow2 = new Dialogflow_V2();

dialogflow2.setConfiguration = async function (clientEmail, privateKey, languageTag, projectId) {
    dialogflow2.accessToken = await dialogflow2.generateAccessToken(clientEmail, privateKey);
    dialogflow2.languageTag = languageTag;
    dialogflow2.projectId = projectId;
    dialogflow2.sessionId = dialogflow2.sessionId ? dialogflow2.sessionId : dialogflow2.guid();

    // set listeners
    Voice.onSpeechStart = (c) => {
        dialogflow2.speechResult = null;
        if (dialogflow2.onListeningStarted) {
            dialogflow2.onListeningStarted(c);
        }
    }

    Voice.onSpeechEnd = (c) => {

        if (dialogflow2.speechResult) {
            dialogflow2.requestQuery(dialogflow2.speechResult[0], dialogflow2.onResult, dialogflow2.onError);
        }

        if (dialogflow2.onListeningFinished) {
            dialogflow2.onListeningFinished(c);
        }
    }

    Voice.onSpeechVolumeChanged = (c) => {
        if (dialogflow2.onAudioLevel) {
            dialogflow2.onAudioLevel(c);
        }
    }

    Voice.onSpeechResults = (result) => {
        if (result.value) {
            dialogflow2.speechResult = result.value;
        }
    }
}

dialogflow2.startListening = function (onResult, onError) {
    dialogflow2.onResult = onResult;
    dialogflow2.onError = onError;

    Voice.start(dialogflow2.languageTag);
}
`
BEFORE:
`/**
 * DIALOGFLOW V2
 */
var dialogflow2 = new Dialogflow_V2();

dialogflow2.setConfiguration = async function (clientEmail, privateKey, languageTag, projectId) {
    dialogflow2.accessToken = await dialogflow2.generateAccessToken(clientEmail, privateKey);
    dialogflow2.languageTag = languageTag;
    dialogflow2.projectId = projectId;
    dialogflow2.sessionId = dialogflow2.sessionId ? dialogflow2.sessionId : dialogflow2.guid();
    
    // set listeners
    Voice.onSpeechStart = () => (c) => dialogflow2.onListeningStarted(c);
    Voice.onSpeechEnd = () => (c) => dialogflow2.onListeningFinished(c);
    Voice.onSpeechError = (err) => dialogflow2.onListeningError(err);
}

dialogflow2.startListening = function (onResult, onError) {

    dialogflow2.subscription = NativeAppEventEmitter.addListener(
        'onSpeechResults',
        (result) => {
            if (result.value) {
                dialogflow2.requestQuery(result.value[0], onResult, onError);
            }

        }
    );

    Voice.start(dialogflow2.languageTag);
}`

It works for me!
